src/lucky_cli/task.cr

```diff
@@ -60,6 +68,123 @@ abstract class LuckyCli::Task
     end
   end
 
+  # Creates a method of `arg_name` that returns the value passed in from the CLI.
+  # The CLI arg position is based on the order in which `positional_arg` is specified
+  # with the first call being position 0, and so on.
+  #
+  # If your arg takes more than one value, you can set `to_end` to true to capture all
+  # args from this position to the end. This will make your `arg_name` method return `Array(String)`.
+  #
+  # * `arg_name` : String - The name of the argument
+  # * `description` : String - The help text description for this option
+  # * `required` : Bool - raise exception if this arg is not passed
+  # * `to_end` : Bool - Capture all args from this position to the end.
+  # * `format` : Regex - The format you expect the args to match
+  macro positional_arg(arg_name, description, required = false, to_end = false, format = nil)
+    {% PARSER_OPTS << arg_name %}
+    @{{ arg_name.id }} : {% if to_end %}Array(String){% else %}String{% end %} | Nil
+
+    def set_opt_for_{{ arg_name.id }}(args : Array(String))
+      {% if to_end %}
+        value = args[@positional_arg_count..-1]
+      {% else %}
+        value = args[@positional_arg_count]?
+      {% end %}
+      {% if format %}
+      matches = value.is_a?(Array) ? value.all?(&.=~({{ format }})) : value =~ {{ format }}
+      if !matches
+        raise "Invalid format for {{ arg_name.id }}. It should match {{ format }}"
+      end
+      {% end %}
+      @{{ arg_name.id }} = value
+      @positional_arg_count += 1
+    end
+
+    def {{ arg_name.id }}
+      {% if required %}
+        if @{{ arg_name.id }}.nil?
```

Comment:
I'm kinda thinking that `positional_arg` that aren't `to_end` should always be required. Otherwise if you have 3 args and the middle one is optional should the second arg be mapped to the third argument? Or did they just forget the second arg?

So I'm thinking we require positional_args

For positonal_args with `to_end` I think it can be optional by checking `empty?` So by default it requires *something* but if set to optional it is ok with an empty array. This would also need a matching spec of course

Thoughts?